spec.emu

```diff
@@ -882,6 +882,297 @@ contributors: Google, Ecma International
+          _year_: an integer,
+        ): an integer
+      </h1>
+      <dl class="header">
+        <dt>description</dt>
+        <dd>
+          It interprets _year_ as arithmetical value in the given _calendar_ and returns the number of months in the corresponding year.
+        </dd>
+      </dl>
+      <p>It performs the following steps when called:</p>
+      <emu-alg>
+        1. Return a calendar-dependent integer as described above.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-temporal-regulatenonisoyearmonth" type="implementation-defined abstract operation">
+      <h1>
+        RegulateNonISOYearMonth (
+          _calendar_: a calendar type that is not *"iso8601"*,
+          _arithmeticalYear_: an integer,
+          _monthCode_: a String,
+          _years_: an integer,
+        ): a Record with fields [[Year]] (an integer) and [[Month]] (an integer)
+      </h1>
+      <dl class="header">
+        <dt>description</dt>
+        <dd>
+          Interprets _isoDate_ as a date in _calendar_, adds _years_, and returns the best-match arithmetic _year_ and _month_ according to the calendar.
+        </dd>
+      </dl>
+      <p>It performs the following steps when called:</p>
+      <emu-alg>
+        1. Let _y2_ be _arithmeticalYear_ + _years_.
+        1. If _calendar_ is *"chinese"*, *"dangi"*, or *"hebrew"*, then
+          1. If _calendar_ is *"chinese"* or *"dangi"*, then
+            1. Let _month_ be MonthCodeToArithmeticalMonth(_calendar_, _y2_, _monthCode_).
+            1. If _month_ is ~invalid~, then
+              1. Let _monthCode_ be CreateMonthCode(ParseMonthCode(_monthCode_).[[Number]], *false*).
+              1. Set _month_ to MonthCodeToArithmeticalMonth(_calendar_, _y2_, _monthCode_).
+              1. Assert: _month_ is not ~invalid~.
+          1. Else,
+            1. Assert: _calendar_ is *"hebrew"*.
+            1. Let _isLeap1_ be *true* if _arithmeticalYear_ is a leap year in the Hebrew calendar, or *false* otherwise.
+            1. Let _isLeap2_ be *true* if _y2_ is a leap year in the Hebrew calendar, or *false* otherwise.
+            1. If _isLeap1_ is *true* and _isLeap2_ is *false*, then
+              1. If _monthCode_ is *"M05L"*, then
+                1. Set _monthCode_ to *"M06"*.
+            1. Else if _isLeap1_ is *false* and _isLeap2_ is *true*, then
+              1. If _monthCode_ is *"M06"*, then
+                1. Set _monthCode_ to *"M05L"*.
+            1. Let _month_ be MonthCodeToArithmeticalMonth(_calendar_, _monthCode_, _y2_).
+            1. Assert: _month_ is not ~invalid~.
+        1. Else,
+          1. Let _month_ be ParseMonthCode(_monthCode_).[[Number]].
+        1. Return the Record { [[Year]]: _y2_, [[Month]]: _month_ }.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-temporal-balancenonisodate" type="implementation-defined abstract operation">
+      <h1>
+        BalanceNonISODate (
```

Comment:
praise: well written.

suggestion: I suppose this could be refactored into BalanceNonISOYearMonth and BalanceNonISODate to avoid the duplication of the `resolvedMonth` balancing. But I don't know if that's better.

Reply:
I changed around the AOs in 6abbe6e3ca1ca220e33f3a62cb474fe631a75495